### PR TITLE
Adding a few macros and functions to capture more error details throw…

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This can probably use some more spit and polish but just throwing it in there as I wrote these quickly this morning to debug some iPhone issues.

I've added a couple of macros and a couple of supporting functions to log a little more info about any errors generated by OpenGL. While overkill to do for every command adding these to more failure sensitive functions can help find issues quicker as it will output the location and name of the constant (where I've added it to the list) of the error being thrown.

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*************************************************************************/
+    /*************************************************************************/
```
Comment (akien-mga) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: Leading spaces here, clang-format will fail.

**Comment by akien-mga** [comment]
`clang-format` failed due to some trailing spaces and use of spaces for indentation: https://travis-ci.org/godotengine/godot/jobs/232026038

You *really* need to set up the precommit hook :]

**Comment by BastiaanOlij** [comment]
ROFL, i though I ran that ..... thing :) Mind you, I was rushing a bit, I kept doing things in 15 - 30 minute blocks over the weekend.

Have you got a good info page somewhere on setting up pre-commit hooks. Might as well find a few minutes to set it up..

**Comment by akien-mga** [comment]
> Have you got a good info page somewhere on setting up pre-commit hooks. Might as well find a few minutes to set it up..

https://github.com/godotengine/godot/blob/master/misc/hooks/README.md

**Comment by karroffel** [comment]
@BastiaanOlij reduz says that he prefers to use external debugging tools for that, so I'll close this.

**Comment by reduz** [comment]
I prefer to use an external opengl debugger like RenderDoc to better be able to tell what's going on than adding those macros everywhere

**Pull request closed without merging**